internal/datastore/spanner/spanner.go

```diff
@@ -98,12 +98,16 @@ func NewSpannerDatastore(database string, opts ...Option) (datastore.Datastore,
 	}
 	ds.RemoteClockRevisions.SetNowFunc(ds.HeadRevision)
 
-	ctx, cancel := context.WithCancel(context.Background())
-	if err := ds.runGC(ctx); err != nil {
-		cancel()
-		return nil, fmt.Errorf(errUnableToInstantiate, err)
+	if config.gcInterval > 0*time.Minute && config.gcEnabled {
```

Comment:
Every GC callsite has the same condition `if datastore.gcInterval > 0*time.Minute && config.gcEnabled`. The `GCEnabled` configuration is currently set to `mysql.GCEnabled(!opts.ReadOnly)`. Why don't we make it instead `mysql.GCEnabled(!opts.ReadOnly && opts.gcInterval > 0*time.Minute)` so that these callsites only need to do `if config.gcEnabled`?

Reply:
Because nothing precludes a caller from also setting the interval manually to 0 time, and we should either raise an error or disable GC in that case (which is, in either case, a check logic)

Reply:
yeah that's what I mean: we can assume `0` as interval should be checked earlier on process start and that it's not a valid value.

Reply:
I tend to still want to have the check within the handlers, to be safe